File: NextToMeAPI/NextToMe.API/Controllers/AuthController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using NextToMe.Common.DTOs;
using NextToMe.Common.Models;
using NextToMe.Services.ServiceInterfaces;

namespace NextToMe.API.Controllers
{
    [ApiController]
    [AllowAnonymous]
    [Route("api/auth")]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;

        public AuthController(IAuthService authService)
        {
            _authService = authService;
        }

        [HttpPost]
        [Route("login")]
        public async Task<LoginResponse> Login(LoginRequest request)
        {
            return await _authService.Login(request);
        }

        [HttpPost]
        [Route("register")]
        public async Task Register(RegisterRequest request)
        {
            await _authService.Register(request);
        }

        [HttpPost]
        [Route("password/reset")]
        public async Task ResetPassword(ResetPasswordRequest request)
        {
            await _authService.ResetPassword(request);
        }

        [HttpPost]
        [Route("password/set")]
        public async Task SetNewPassword(SetNewPasswordRequest request)
        {
            await _authService.SetNewPassword(request);
        }

        [HttpPost]
        [Route("confirm")]
        public async Task ConfirmEmail(EmailConfirmRequest request)
        {
            await _authService.ConfirmEmail(request);
        }

        [HttpPost]
        [Route("refresh")]
        public async Task<LoginResponse> RefreshToken(RefreshTokenRequest request)
        {
            return await _authService.RefreshToken(request);
        }
    }
}

File: NextToMeUI/api/AuthController.js
import APIController from "./ApiController";

class AuthController extends APIController {
  constructor() {
    super();
  }

  /**  
    @param {Object} user_info
      @param {String} login
      @param {String} password
  */
  async login(user_info) {
    const loginResponse = await this.request("post", "/auth/login", user_info);
    this.updateAuthHeader(loginResponse.data.accessToken);
    const { data } = loginResponse;
    return data;
  }

  /**  
    @param {Object} user_info
      @param {String} login
      @param {String} redirectUrl - url, с окном для ввода пароля
  */
  async signup(user_info) {
    const signupResponse = await this.request(
      "post",
      "/auth/register",
      user_info
    );
    return signupResponse;
  }

  /**  
    @param {Object} user_info
      @param {String} userId
      @param {String} code 
      @param {String} password 
  */
  async confirm(user_info) {
    const confirmResponse = await this.request(
      "post",
      "/auth/confirm",
      user_info
    );
    return confirmResponse;
  }

  /**  
    @param {Object} user_info
      @param {String} login
      @param {String} redirectUrl
  */
  async resetPassword(user_info) {
    const resetStatus = await this.request(
      "post",
      "/auth/password/reset",
      user_info
    );
    return resetStatus;
  }

  /**  
    @param {Object} user_info
      @param {String} userId
      @param {String} code
      @param {String} newPassword
  */
  async setNewPassword(user_info) {
    const setStatus = await this.request(
      "post",
      "/auth/password/set",
      user_info
    );
    return setStatus;
  }
}

export default new AuthController();
